geonode/layers/templates/layers/layer_detail.html

```diff
@@ -293,6 +293,9 @@ <h4 class="modal-title" id="myModalLabel">{% trans "Download Layer" %}</h4>
                               {% for link in links_download %}
                               <li><a href="{{ link.url }}" target="_blank">{% trans link.name %}</a></li>
                               {% endfor %}
+                              {% if 'geonode.qgis_server' in INSTALLED_APPS %}
+                                <li><a href="{% url 'qgis_server:download-qlr' layername=resource.name %} " target="_blank">QGIS Layer file</a></li>
+                              {% endif %}
```

Comment:
No, no need to change this file. You need to add a new LINK in the database, and then it will automatically come with the line 294:
```
                              {% for link in links_download %}
                                <li><a href="{{ link.url }}" target="_blank">{% trans link.name %}</a></li>
                                {% endfor %}
```

And then your link will be added automatically in the `Reference` section of that page: http://staging.geonode.kartoza.com/layers/geonode:tmpeqfzkt/metadata_detail

Create the link in the post_save signal from QGIS Server app, exactly like we are doing for ZIP, WMS, WFS, ...